### PR TITLE
haskell-packages: Add minimal benchDepends support

### PR DESCRIPTION
**Pull request by expipiplus1**
See also: https://github.com/NixOS/cabal2nix/pull/262

cc @peti

### CODE REVIEW

**Comment by domenkozar** [comment]
Looks harmless :)

File: pkgs/development/haskell-modules/generic-builder.nix [hunk 1]
```diff
@@ -13,6 +13,7 @@
 , configureFlags ? []
 , description ? ""
 , doCheck ? !isCross && (stdenv.lib.versionOlder "7.4" ghc.version)
+, doBench ? false
```
Comment (peti) [inline pkgs/development/haskell-modules/generic-builder.nix]: This option suggests to users that the build would run (or compile) the package's benchmarks, but in fact it does not. That feels a little counter-intuitive and misleading.

File: pkgs/development/haskell-modules/generic-builder.nix [hunk 2]
```diff
@@ -38,6 +39,7 @@
 , passthru ? {}
 , pkgconfigDepends ? [], libraryPkgconfigDepends ? [], executablePkgconfigDepends ? [], testPkgconfigDepends ? []
 , testDepends ? [], testHaskellDepends ? [], testSystemDepends ? []
+, benchDepends ? []
```
Comment (peti) [inline pkgs/development/haskell-modules/generic-builder.nix]: As mentioned elsewhere, this option should be called "benchmarkDepends", IMHO, for consistency.

File: pkgs/development/haskell-modules/generic-builder.nix (hunk 1 above)
Comment (expipiplus1) [inline pkgs/development/haskell-modules/generic-builder.nix]: would `withBenchmarkDepends` be better? I'm not sure when we'd want to run the benchmarks in a nix build. My use case here is to make the benchmark dependencies available in the shell environment.

File: pkgs/development/haskell-modules/generic-builder.nix (hunk 2 above)
Comment (expipiplus1) [inline pkgs/development/haskell-modules/generic-builder.nix]: gotcha, I'll open another PR.

**Pull request merged**